### PR TITLE
Events for user preferences

### PR DESCRIPTION
**Pull request by ainouzgali**

### CODE REVIEW

File: apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts [hunk 1]
```diff
@@ -28,6 +31,12 @@ export class GetSubscriberPreference {
       true
     );
 
+    this.analyticsService.track('Fetch User Preferences - [Notification Center]', command.organizationId, {
+      _organization: command.organizationId,
+      _subscriber: command.subscriberId,
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/get-subscriber-preference/get-subscriber-preference.usecase.ts:36]:
```suggestion
      _subscriber: subscriber._id,
```

We should use only the internal id here, or just remove it all together

File: apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts [hunk 2]
```diff
@@ -75,6 +78,13 @@ export class UpdateSubscriberPreference {
       }
     );
 
+    this.analyticsService.track('Update User Preference - [Notification Center]', command.organizationId, {
+      _organization: command.organizationId,
+      _subscriber: command.subscriberId,
+      _template: command.templateId,
+      ...updatePayload,
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts:85]: No need to have the payload here

File: apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts [hunk 3]
```diff
@@ -75,6 +78,13 @@ export class UpdateSubscriberPreference {
       }
     );
 
+    this.analyticsService.track('Update User Preference - [Notification Center]', command.organizationId, {
+      _organization: command.organizationId,
+      _subscriber: command.subscriberId,
```
Comment (scopsy) [inline apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts:83]: Same comment for subscriber id here :)

File: apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts (hunk 2 above)
Comment (ainouzgali) [reply apps/api/src/app/widgets/usecases/update-subscriber-preference/update-subscriber-preference.usecase.ts:85]: Wouldn't we want to know what was enabled/disabled?

**Review by scopsy — APPROVED** [review]

**Pull request merged**